### PR TITLE
SSCS-10295 adjournment interpreter requirements

### PR DESCRIPTION
**Pull request by gregorio-massara**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java [hunk 1]
```diff
@@ -690,4 +690,41 @@ void testIsPoOfficerAttendingOverrideNull() {
 
         assertThat(result).isFalse();
     }
+
+    @DisplayName("Interpreter requirements selected during Adjournment is used in next hearing request (German interpreter required) ")
+    @Test
+    void testAdjournInterpreterRequirementsUsedInNextHearingRequestGermanLanguage() {
+        SscsCaseData caseData = SscsCaseData.builder()
+            .adjournCaseInterpreterRequired("Yes")
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java:698]: Yes.getValue()

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java [hunk 2]
```diff
@@ -690,4 +690,41 @@ void testIsPoOfficerAttendingOverrideNull() {
...
+            .adjournCaseInterpreterLanguage("German")
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java:699]: Make constant

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java [hunk 3]
```diff
@@ -690,4 +690,41 @@ void testIsPoOfficerAttendingOverrideNull() {
...
+        String language = HearingsDetailsMapping.adjournCaseInterpreterLanguage(caseData);
+
+        assertThat(language).isEqualTo("German");
+    }
+
+    @DisplayName("Interpreter requirements selected during Adjournment is used in next hearing request (no interpreter required) ")
+    @Test
+    void testAdjournInterpreterRequirementsUsedInNextHearingRequestNoInterpreter() {
+        SscsCaseData caseData = SscsCaseData.builder()
+            .adjournCaseInterpreterRequired("No")
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java:711]: NO.getValue()

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java [hunk 4]
```diff
@@ -690,4 +690,41 @@ void testIsPoOfficerAttendingOverrideNull() {
...
+            .build();
+
+        String language = HearingsDetailsMapping.adjournCaseInterpreterLanguage(caseData);
+
+        assertThat(language).isEqualTo("no interpreter required");
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java:716]: String constant

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingValuesMappingTest.java [hunk 5]
```diff
@@ -106,6 +106,8 @@ public void setUp() {
             .issueCode(ISSUE_CODE)
             .urgentCase("Yes")
             .adjournCaseCanCaseBeListedRightAway("Yes")
+            .adjournCaseInterpreterRequired("Yes")
+            .adjournCaseInterpreterLanguage("Bulgarian")
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingValuesMappingTest.java:110]: Constant

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingValuesMappingTest.java [hunk 6]
```diff
@@ -241,6 +243,7 @@ void shouldMapServiceHearingValuesSuccessfully() throws InvalidMappingException
         assertNull(serviceHearingValues.getVocabulary());
         assertEquals(List.of(FACE_TO_FACE), serviceHearingValues.getHearingChannels());
         assertEquals(true, serviceHearingValues.isCaseInterpreterRequiredFlag());
+        assertEquals("Bulgarian", serviceHearingValues.getAdjournCaseInterpreterLanguage());
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/ServiceHearingValuesMappingTest.java:246]: Please replace all org.junit.jupiter.api.Assertions with org.assertj.core.api.Assertions
ie assertEquals becomes assertThat(...).isEqualTo()

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java [hunk 7]
```diff
@@ -64,6 +65,7 @@ public static HearingDetails buildHearingDetails(HearingWrapper wrapper, Referen
             .hearingIsLinkedFlag(isCaseLinked(caseData))
             .amendReasonCodes(OverridesMapping.getAmendReasonCodes(caseData))
             .hearingChannels(HearingsChannelMapping.getHearingChannels(caseData))
+            .interpreterLanguage(adjournCaseInterpreterLanguage(caseData))
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java:68]: Does this not need a check to see if the case is adjourned or not?

File: config/owasp/suppressions.xml [hunk 8]
```diff
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.3.xsd">
+  <suppress until = "2022-10-15">
+    <notes><![CDATA[
+        file name: jackson-databind-2.11.4.jar
+        ]]></notes>
+    <packageUrl regex="true">^pkg:maven/com\.fasterxml\.jackson\.core/jackson\-databind@.*$</packageUrl>
+    <cve>CVE-2022-42003</cve>
+    <cve>CVE-2022-42004</cve>
+  </suppress>
```
Comment (bensonlievoco) [inline config/owasp/suppressions.xml:10]:
```suggestion
  please rebase your pr, that should be fixed 
```

Comment (amoslv1) [reply config/owasp/suppressions.xml:10]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java [hunk 9]
```diff
@@ -212,4 +214,15 @@ public static boolean isPoOfficerAttending(@Valid SscsCaseData caseData) {
...
     }
+
+    public static String adjournCaseInterpreterLanguage(@Valid SscsCaseData caseData) {
+        if (isNull(caseData.getAdjournCaseInterpreterRequired())) {
+            return caseData.getLanguagePreference().getCode();
+        }
+        if (isYes(caseData.getAdjournCaseInterpreterRequired()) && nonNull(caseData.getAdjournCaseInterpreterLanguage())) {
+            return caseData.getAdjournCaseInterpreterLanguage();
+        }
+        return "no interpreter required";
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java:225]: Magic string - should this not be a constant? And checking that this is what is actually required to be returned

**Review by hgulv1 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java (hunk 9 above)
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java]: Made string into a constant. 
When demoing this to Makeda, she was happy with this behaviour.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/service/hearingvalues/ServiceHearingValues.java [hunk 10]
```diff
@@ -97,4 +97,7 @@ public class ServiceHearingValues {
     private String hmctsServiceID;
 
     private boolean caseInterpreterRequiredFlag;
+
+    private String adjournCaseInterpreterLanguage;
```
Comment (TeaWithLucas) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/service/hearingvalues/ServiceHearingValues.java:101]: THis should not be added

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingDetails.java [hunk 11]
```diff
@@ -58,4 +58,6 @@ public class HearingDetails {
 
     private List<HearingChannel> hearingChannels;
 
+    private String interpreterLanguage;
```
Comment (TeaWithLucas) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingDetails.java:61]: THis should not be added

**Comment by pk2453** [comment]
New branch SSCS-10295-1 is created for this.

**Pull request closed without merging**